src/config.cpp

```diff
@@ -59,6 +66,106 @@ void parse_uint32(void* value_ptr, const char* txt) {
     cfgval->set = true;
 }
 
+// Helper function to trim whitespace and remove quotations
+std::string trim(const std::string& str) {
+    size_t first = str.find_first_not_of(" \t\n\""); // Also include quotes
+    size_t last = str.find_last_not_of(" \t\n\"");
+
+    // Check for valid range
+    if (first == std::string::npos || last == std::string::npos) {
+        return str; // Empty or only whitespace/quotes
+    }
+
+    // Extract the trimmed and unquoted substring
+    std::string trimmed = str.substr(first, (last - first + 1));
+
+    // Check if the trimmed string still starts and ends with quotes
+    if (trimmed.front() == '"' && trimmed.back() == '"') {
+        return trimmed.substr(1, trimmed.size() - 2); // Remove the quotes
+    } else {
+        return trimmed;
+    }
+}
+
+void read_config_file(std::unordered_map<std::string, std::string>& umap,
+                      std::ifstream& config_stream) {
+
+    std::string line;
+    while (std::getline(config_stream, line)) {
+        // Ignore comments and empty lines
+        if (line.empty() || line[0] == '#') {
+            continue;
+        }
+        line = trim(line);
+        // Check for section headers
+        if (line[0] == '[' && line[line.size() - 1] == ']') {
+            continue;
+        } else {
+            // Parse key-value pairs
+            size_t pos = line.find('=');
+            if (pos != std::string::npos) {
+                std::string key = trim(line.substr(0, pos));
+                std::string value = trim(line.substr(pos + 1));
+                // Store values (if any)
+                if (value != "") {
+                    umap[key] = value;
```

Comment:
```suggestion
                    umap[key] = value;
                    cvk_info_fn("%s = %s", key.c_str(), value.c_str());
```

Reply:
Maybe better for the debug log level? This could be very noisy. Maybe it makes sense to introduce a 'config' log group otherwise?

Reply:
I don't see it being that noisy, but I don't mind, as long as we have them it's fine with me. I agree a log group would make sense.